app/models/concerns/counter_cache_fix.rb

```diff
@@ -14,9 +14,16 @@ def update_counter_caches
         if key =~ /_id/
           association = self.association(key.sub(/_id$/, '').to_sym)
           if association.options[ :counter_cache ]
-            counter_name = self.class.name.underscore.split("/")[0].pluralize.to_sym
-            association.klass.reset_counters(old_value, counter_name) if old_value
-            association.klass.reset_counters(new_value, counter_name) if new_value
+
+            if association.options[:inverse_of].is_a?(Symbol)
+              counter_name = association.options[:inverse_of]
+            else
+              counter_name = self.class.name.underscore.split("/")[0].pluralize.to_sym
+            end
+
+            association_name = counter_name.to_s.sub(/_count$/, "").to_sym
```

Comment:
Sorry :bow: , I can't figure out right now what made you add this? A comment here would be useful.